src/main/java/org/elasticsearch/index/query/SpanTermQueryBuilder.java

```diff
@@ -19,83 +19,71 @@
 
 package org.elasticsearch.index.query;
 
-import org.elasticsearch.common.xcontent.XContentBuilder;
-
-import java.io.IOException;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.spans.SpanTermQuery;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.lucene.BytesRefs;
+import org.elasticsearch.index.mapper.MapperService;
 
 /**
- *
+ * A Span Query that matches documents containing a term.
+ * @see SpanTermQuery
  */
-public class SpanTermQueryBuilder extends BaseQueryBuilder implements SpanQueryBuilder, BoostableQueryBuilder<SpanTermQueryBuilder> {
-
-    private final String name;
-
-    private final Object value;
-
-    private float boost = -1;
-
-    private String queryName;
-
+public class SpanTermQueryBuilder extends BaseTermQueryBuilder<SpanTermQueryBuilder> implements SpanQueryBuilder {
+    /** @see BaseTermQueryBuilder#BaseTermQueryBuilder(String, String) */
     public SpanTermQueryBuilder(String name, String value) {
-        this(name, (Object) value);
+        super(name, (Object) value);
     }
-
+    /** @see BaseTermQueryBuilder#BaseTermQueryBuilder(String, int) */
     public SpanTermQueryBuilder(String name, int value) {
-        this(name, (Object) value);
+        super(name, (Object) value);
     }
-
+    /** @see BaseTermQueryBuilder#BaseTermQueryBuilder(String, long) */
     public SpanTermQueryBuilder(String name, long value) {
-        this(name, (Object) value);
+        super(name, (Object) value);
     }
-
+    /** @see BaseTermQueryBuilder#BaseTermQueryBuilder(String, float) */
     public SpanTermQueryBuilder(String name, float value) {
-        this(name, (Object) value);
+        super(name, (Object) value);
     }
-
+    /** @see BaseTermQueryBuilder#BaseTermQueryBuilder(String, double) */
     public SpanTermQueryBuilder(String name, double value) {
-        this(name, (Object) value);
+        super(name, (Object) value);
     }
-
-    private SpanTermQueryBuilder(String name, Object value) {
-        this.name = name;
-        this.value = value;
+    /** @see BaseTermQueryBuilder#BaseTermQueryBuilder(String, Object) */
+    public SpanTermQueryBuilder(String name, Object value) {
+        super(name, value);
     }
 
-    @Override
-    public SpanTermQueryBuilder boost(float boost) {
-        this.boost = boost;
-        return this;
+    SpanTermQueryBuilder() {
```

Comment:
can you make this public? I think it will need to be public for serialization at some point